desc/magnetic_fields.py

```diff
@@ -198,13 +430,23 @@ def compute_magnetic_field(self, coords, params=None, basis="rpz"):
         field : ndarray, shape(N,3)
             scaled magnetic field at specified points
         """
+        from desc.coils import Coil
+
         if params is None:
             params = [None] * len(self._fields)
         if isinstance(params, dict):
             params = [params]
         B = 0
         for i, field in enumerate(self._fields):
-            B += field.compute_magnetic_field(coords, params[i % len(params)], basis)
+            B += (
+                field.compute_magnetic_field(
+                    coords, params[i % len(params)], basis, grid=grid
+                )
+                if isinstance(field, Coil)
+                else field.compute_magnetic_field(
+                    coords, params[i % len(params)], basis
+                )
```

Comment:
Since the base class has grid as part of the API we should update the others as well, which will avoid the need for this logic